### PR TITLE
Media Query Mixins

### PR DESCRIPTION
**Pull request by AlecRaeside**

### CODE REVIEW

File: core/mixins/_media-queries.scss [hunk 1]
```diff
@@ -35,189 +39,39 @@
 /* Simple minimum and maximum width/height values
    ========================================================================== */
```
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: This type of comment is reserved for when you have a huge amount of CSS so you can easily break it up but most of the framework doesn't need them as we're very modular. So just the standard will do e.g. https://github.com/westfieldlabs/scally/blob/master/core/mixins/_convert-px-to-em-rem.scss#L31-L33.

File: core/mixins/_media-queries.scss [hunk 2]
```diff
@@ -10,15 +10,19 @@
  * https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Media_queries.
```
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: Let's remove this. And add the `@demo` URL as per: https://github.com/westfieldlabs/scally/blob/master/core/mixins/_convert-px-to-em-rem.scss#L23-L24.

File: core/mixins/_media-queries.scss (hunk 1 above)
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: I'll be updating the [CSS Guidelines](https://github.com/westfield/css_guidelines) once Scally is complete as we have some new authoring guidelines mainly around how we write comments.

File: core/mixins/_media-queries.scss [hunk 3]
```diff
@@ -35,189 +39,39 @@
...
- */
-
-@mixin respond-min($value, $height: false) {
-
-  // Height
-  @if $height {
-    @media (min-height: ($value / 16)+em) {
-      @content;
-    }
+@mixin respond-to($value, $axis: "width", $limit: "min") {
```
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: I'm thinking `$limit` before `$axis` as I reckon `$limit` will get used more? So we can write mq's like:

```
@include respond-to(lap, max) { ... }
```

File: core/mixins/_media-queries.scss [hunk 4]
```diff
@@ -10,15 +10,19 @@
...
+   @include respond-to(500) {
      .foo {background: red;}
    }
 
-   @include respond-max($palm-end) {
+   @include respond-to(palm, $limit: "max") {
+     .foo {background: red;}
+   }
+
+   @include respond-to(500, $axis: "height") {
```
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: Couldn't this be the less verbose version?

```
@include respond-to(500, height) {
```

But see comment below about the order of these arguments.

File: core/mixins/_media-queries.scss [hunk 5]
```diff
@@ -35,189 +39,39 @@
...
-  // Height
-  @if $height {
-    @media (min-height: ($value / 16)+em) {
-      @content;
-    }
+@mixin respond-to($value, $axis: "width", $limit: "min") {
+  // If value is a named breakpoint
+  @if type-of($value) == "string" {
+    $value: map-get($breakpoints, $value);
   }
```
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: Can we include `@warn`'s?

File: core/mixins/_media-queries.scss (hunk 3 above)
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: Not sure about the name `$axis` as straight away I'm thinking `x`, `y`, and `z` as the values.

File: core/mixins/_media-queries.scss (hunk 5 above)
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: As it fails if you enter a string that isn't one of the breakpoints.

File: core/mixins/_media-queries.scss [hunk 6]
```diff
@@ -35,189 +39,39 @@
...
+    $min-value: map-get($breakpoints, $min-value);
 
-// Accepts a minimum and maximum width/height range
-@mixin respond-range($min-value, $max-value, $height: false) {
-
-  // Height range
-  @if $height {
-    @media (min-height: ($min-value / 16)+em) and (max-height: ($max-value / 16)+em) {
-      @content;
+    // If max-value is omitted, use the next named breakpoint after the min-value breakpoint
```
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: Not sure about this, I'm leaning towards to just rendering an error if `$max-vaue` is omitted?

Comment (AlecRaeside) [inline core/mixins/_media-queries.scss]: This is included to cover the functionality of the `respond-range-fixed` mixin in CC https://github.com/AlecRaeside/customer_console/blob/master/app/assets/stylesheets/partials/utils/_mixins.scss#L768 which is used in a couple of place in CC

File: core/mixins/_media-queries.scss [hunk 7]
```diff
@@ -35,189 +39,39 @@
...
+    @if $max-value == 0 {
+      $breakpoint-values: map-values($breakpoints);
+      $breakpoint-index:index($breakpoint-values, $min-value);
```
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: missing space after `$breakpoint-index:`.

File: core/mixins/_media-queries.scss (hunk 6 above)
Comment (chris-pearce) [inline core/mixins/_media-queries.scss]: Ah right

**Comment by AlecRaeside** [comment]
I've updated after code review. I've also moved logic to get the breakpoint values from breakpoint names into the to-em function. This has made the mixins much more DRY as the name validity checks are not repeated everywhere. @chris-pearce

**Comment by chris-pearce** [comment]
@AlecRaeside nice work, looking very clean, just some authoring things, we're doing 2 line breaks now between the partial heading and intro block: https://github.com/AlecRaeside/scally/blob/refactor-mixins/core/mixins/_media-queries.scss#L4. And same spacing between the style blocks e.g. https://github.com/westfieldlabs/scally/blob/master/core/mixins/_gradients.scss. Also we're wrapping at 100 characters, I've got this in my Sublime Text preferences file:

```
"rulers":
 [
   100
 ],
```

This will all be documented properly in the new Guidelines once Scally is done.

**Comment by AlecRaeside** [comment]
added spacing. I've added that config cheers.  line 66 i exactly 100 chars long

**Pull request merged**